### PR TITLE
Dsable ATT test timeouts when debugger is attached

### PR DESCRIPTION
**Pull request by dbelcham**
Solves #4402

### CODE REVIEW

File: src/NServiceBus.AcceptanceTesting/Support/ScenarioRunner.cs [hunk 1]
```diff
@@ -201,10 +201,13 @@ static void PrintSettings(IEnumerable<KeyValuePair<string, object>> settings)
                     var maxTime = runDescriptor.Settings.TestExecutionTimeout ?? TimeSpan.FromSeconds(90);
                     while (!done() && !cts.Token.IsCancellationRequested)
                     {
-                        if (DateTime.UtcNow - startTime > maxTime)
+                        if (!Debugger.IsAttached)
```
Comment (danielmarbach) [inline src/NServiceBus.AcceptanceTesting/Support/ScenarioRunner.cs]: Do you know is this property cached or will we suffer perf hit of this call?

Comment (dbelcham) [inline src/NServiceBus.AcceptanceTesting/Support/ScenarioRunner.cs]: Looking at the `Debugger` source it looks like the property has a getter only. I can't see anywhere it's being set though so I'm not 100% sure it's "cached" or not

Comment (timbussmann) [inline src/NServiceBus.AcceptanceTesting/Support/ScenarioRunner.cs]: at least it doesn't look like the IsAttached property is cached anywhere when I look at it's usages in referencesource.

**Comment by timbussmann** [comment]
I think you didn't address the timeouts applied on the `When` conditions but only for the `Done` condition. See `ScenarioRunner.ExecuteWhens`.

**Comment by danielmarbach** [comment]
For me that is ok since mostly we run into done condition issue. Sep. PR?

Tim Bussmann <notifications@github.com> schrieb am Mo. 23. Jan. 2017 um
15:07:

**Pull request merged**